tests/kokkos/kokkos_utils_tests.cpp

```diff
@@ -544,4 +544,230 @@ TEST_CASE("subviews") {
   }
 }
 
+TEST_CASE("multi-slice subviews") {
```

Comment:
You should add some tests that catch errors. Since we used `assert` instead of exceptions, we can't use `REQUIRE_THROWS`. 

Perhaps we could write a separate test with the WILL_FAIL option, and run one bad op in there. We can't test more since the assert crashes at the first error.. but maybe we can put N bad calls and randomly choose one to execute. Over time, we'd have some sort of coverage. Of course, this test can only be added in debug mode, since asserts are not executed in release.

@jgfouca @tcclevenger do you have other ideas on how to test that asserts do indeed work?

Reply:
@bartgol , I recommend using the EKAT assert macros in ekat_assert.hpp. 
```
EKAT_ASSERT(false); // throws a logic_error
```

That would allow us to use REQUIRE_THROWS. This won't work for `EKAT_KERNEL_ASSERT` since that's just a Kokkos::abort.